prometheus/metrics.go

```diff
@@ -149,6 +149,14 @@ func getServiceMetrics(api v1.API, q *ServiceMetricsQuery) Metrics {
 	clustername := config.Get().ExternalServices.Istio.IstioIdentityDomain
 	destService := fmt.Sprintf("destination_service=\"%s.%s.%s\"", q.Service, q.Namespace, clustername)
 	srcService := fmt.Sprintf("source_service=\"%s.%s.%s\"", q.Service, q.Namespace, clustername)
+	if q.Service == "unknown" {
+		destService = "destination_service=\"unknown\""
+		srcService = "source_service=\"unknown\""
+		if q.Namespace != "all" {
+			destService += fmt.Sprintf(",source_service=~\".*\\\\.%s\\\\.%s\"", q.Namespace, clustername)
+			srcService += fmt.Sprintf(",destination_service=~\".*\\\\.%s\\\\.%s\"", q.Namespace, clustername)
+		}
+	}
```

Comment:
I don't think it's the case that 'unknown' is ever a destination but I guess that's not really relevant.   What I'm not really sure about is the special handling for 'all'.   Is this the first time 'all'  is introduced into the metrics api?  I'm not sure we should be leaking 'all', which is a special case in the graph handler but I thought was isolated to the graph handling.

Reply:
By the way, what happens when namespace is "all" but not in the case of "unknown"? Are we going to write `destination_service="rating.all.cluster"` ??

Reply:
@jshaughn I changed it from "all" to "unknown".
@jotak That is what was going to happen: `destination_service="rating.all.cluster"`. But now that I changed it to "unknown"... well...  It'd be `destination_service="rating.unknown.cluster"`. I guess it shoudn't be a problem. If we know the service, we should also be able to know the namespace, since it's part of the prometheus metric.